### ISSUE TITLE
Found a bug in error handling of parsing of incorrectly formatted rpc-url

### ISSUE THREAD

**Issue by laudiacay** [foundry-rs/foundry#3407]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (359dd77 2022-09-15T00:04:51.833948Z)

### What command(s) is the bug in?

forge create

### Operating System

macOS (Apple Silicon)

### Describe the bug

I accidentally ran forge create with a typo'd up `--rpc-url`, the command looked something like this:
`forge create     --rpc-url="= https://mainnet.infura.io/v3/MYAPIKEY"     --mnemonic="secretSECRETsecretSECRET" BlockTime `

and I found a sneaky misbehaving `unwrap` that you may want to get rid of! the backtrace was completely unhelpful so I left it out.
```
[⠆] Compiling...
No files changed, compilation skipped
The application panicked (crashed).
Message:  called `Result::unwrap()` on an `Err` value: 
Invalid provider url

Context:
- Error #0: builder error: relative URL without a base
- Error #1: relative URL without a base
Location: /Users/runner/work/foundry/foundry/common/src/provider.rs:31

This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
[1]    17951 abort      forge create   BlockTime
```

**Comment by DaniPopes** [comment]
This was addressed in #3314

**Comment by mattsse** [comment]
yeh this was improved in #3314 and will no longer panic and display a nicer error with the erroneous URL.
